packages/utils/src/display.ts

```diff
@@ -73,8 +73,11 @@ export function stringify(
     })
   }
 
+  // Prevents infinite loop https://github.com/vitest-dev/vitest/issues/7249
+  const MAX_DEPTH = Number.MAX_SAFE_INTEGER
+  const nextMaxDepth = maxDepth > MAX_DEPTH ? MAX_DEPTH : Math.floor(maxDepth / 2)
   return result.length >= MAX_LENGTH && maxDepth > 1
-    ? stringify(object, Math.floor(maxDepth / 2))
+    ? stringify(object, nextMaxDepth, { maxLength, ...options })
```

Comment:
i noticed that maxLength and options were not being sent to recursive child calls before so i added it here